homeassistant/components/vacuum/__init__.py

```diff
@@ -0,0 +1,287 @@
+        """Map services to methods on VacuumDevice."""
+        method = SERVICE_TO_METHOD.get(service.service)
+        if not method:
+            return
+
+        target_vacuums = component.async_extract_from_service(service)
+        params = dict(service.data)
+        _LOGGER.debug('Service call to %s: %s (%s)',
+                      target_vacuums, method, params)
+
+        update_tasks = []
+        for vacuum in target_vacuums:
+            yield from getattr(vacuum, method['method'])(**params)
+            if not vacuum.should_poll:
+                continue
+
+            update_coro = hass.async_add_job(
+                vacuum.async_update_ha_state(True))
+            if hasattr(vacuum, 'async_update'):
+                update_tasks.append(update_coro)
+            else:
+                yield from update_coro
+
+        if update_tasks:
+            yield from asyncio.wait(update_tasks, loop=hass.loop)
+
+    for service in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[service].get(
+            'schema', VACUUM_SERVICE_SCHEMA)
+        hass.services.async_register(
+            DOMAIN, service, async_handle_vacuum_service,
+            descriptions.get(service), schema=schema)
+
+    return True
+
+
+class VacuumDevice(ToggleEntity):
+    """Representation of a vacuum cleaner robot."""
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner features that are supported."""
+        return 0
+
+    @property
+    def state(self):
+        """State of the vacuum cleaner."""
+        return STATE_UNKNOWN
+
+    @property
+    def status(self):
+        """Return the status of the vacuum cleaner."""
+        return None
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return None
+
+    @property
+    def battery_icon(self):
+        """Return the battery icon for the vacuum cleaner."""
+        charging = False
+        if self.status is not None:
+            charging = 'charg' in self.status.lower()
+        return icon_for_battery_level(
+            battery_level=self.battery_level, charging=charging)
+
+    @property
+    def fanspeed(self):
+        """Return the fan speed of the vacuum cleaner."""
+        return None
+
+    @property
+    def fanspeed_list(self) -> list:
+        """Get the list of available fan speed steps of the vacuum cleaner."""
+        return []
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the vacuum cleaner."""
+        data = {}
+
+        if self.status is not None:
+            data['vacuum_status'] = self.status
+
+        if self.battery_level is not None:
+            data['vacuum_battery'] = self.battery_level
+            data['vacuum_battery_icon'] = self.battery_icon
+
+        if self.fanspeed is not None:
+            data['vacuum_fanspeed'] = self.fanspeed
+            data['vacuum_fanspeed_list'] = self.fanspeed_list
+
```

Comment:
not hidden but filtered out when rendering the state attributes table in the more-info view. 
You can access them in `{{states.vacuum.robot_aspirador.attributes.vacuum_status}}` and so.

The extra complexity is because of the filter, to not get false positives.